server/controllers/item-controller.js

```diff
@@ -0,0 +1,258 @@
+    }).catch(err => {
+      console.error(`Hack_avengers - caught error in 'getItemsById': ${err}`);
+      console.error(err);
+      return err;
+    });
+  };
+}
+
+export const createItem = (schemaName, name) => {
+  return (req, res) => {
+    const body = req.body;
+    const id = req.params.id;
+
+    // patient
+    if (!body) {
+      return res.status(400).json({
+        success: false,
+        error: `You must provide an ${name}.`,
+      });
+    }
+
+    let item;
+    if (name === "exam") {
+      item = new Exam({
+        image: body.image,
+        score: body.score,
+        examInfo: body.examInfo,
+        date: body.date,
+        keyFindings: body.keyFindings,
+        patient: id
+      }); //create new record for exam
+      console.log(item)
+    } else {
+      item = new schemaName(body); //create new record
+    }
+
+
+
+    if (!item) {
+      console.error(`Hack_avengers - 400 in create${name}: '${name}' is malformed.`);
+      return res.status(400).json({
+        success: false,
+        message: `'${name}' is malformed`,
+      });
+    }
+
+
+    return item
+      .save()
+      .then(() => {
+        console.error(`Hack_avengers - 201 in 'createItem': ${name} created!`);
+        return res.status(201).json({
+          success: true,
+          id: item._id,
+          message: `${name} created!`,
+        });
+      })
+      .catch(err => {
+        console.error(`Hack_avengers - caught error in 'createItem'`);
+        Object.keys(err.errors).forEach(errorKey => {
+          console.error(`Hack_avengers ERROR for: ${errorKey}`);
+          console.error(
+            `Hack_avengers => ${((err.errors[errorKey] || {}).properties || {}).message
+            }`,
+          );
+        });
+        return res.status(400).json({
+          success: false,
+          error: err.errors,
+          message: err.errors.name,
+        });
+      });
+  };
+}
+
+export const updateItem = (schemaName, name) => {
+  return async (req, res) => {
+    const body = req.body;
+    const arryOfItem = [body]
+    if (!body) {
+      console.error(`Hack_avengers - 400 in 'update${name}': You must provide ${name} to update.`);
+      return res.status(400).json({
+        success: false,
+        error: `You must provide ${name} to update.`,
+      });
+    }
+
+    const itemForUpdate = { ...arryOfItem }[0]
+
+
+    try {
+      await schemaName.findOneAndUpdate(name === "exam" ? { patient: req.params.id } : { _id: req.params.id }, itemForUpdate);
+    } catch (err) {
+      console.error(`Hack_avengers - caught error in 'update${name}': ${err}`);
+      console.error(err);
+      return res.status(400).json({
+        success: false,
+        error: err,
+      });
+    }
+
+    console.log(`Hack_avengers - 200 in 'update${name}': ${name} updated!`);
+    return res.status(200).json({
+      success: true,
+      id: req.params.id,
+      message: `${name} updated!`,
+    });
+  };
+}
+
```

Comment:
I feel like its great to create a new schema name and pass it to deleteItem from the route. or find another means of deleting. It works but the controller won't be reusable any longer.